### PR TITLE
Add fonctionnality groupBySeries and groupById. Height of cell auto sized

### PR DESCRIPTION
**Pull request by GuillaumeSmaha**
groupBySeries permit to aggregate all elements in the serie to the same row
groupById permit to aggregate all rows with the same id to an uniq row. groupById can ony be activated if groupBySeries = true.

groupByIdDrawAllTitles permit to draw all titles when the option groupById is enabled. Else draw only the first title

### CODE REVIEW

**Comment by lutze** [comment]
@GuillaumeSmaha Do you have an example of this working that I could have a look at?

**Comment by GuillaumeSmaha** [comment]
Yes, I create 3 examples on JSFiddle :

Group by series : http://jsfiddle.net/Mouxon/25esG/
(plain page : http://fiddle.jshell.net/Mouxon/25esG/show/)

Group by series and ID with the first title : http://jsfiddle.net/Mouxon/9tTVc/
(plain page : http://fiddle.jshell.net/Mouxon/9tTVc/show/)

Group by series and ID with all titles : http://jsfiddle.net/Mouxon/4v9VW/
(plain page : http://fiddle.jshell.net/Mouxon/4v9VW/show/)

**Comment by lutze** [comment]
When you say "with the first title" or "all titles" here, I'm seeing it display either the name values in the first series, or the name values in all series. Is that the intent? 

Comments by features: 
1. groupBySeries 
   1. I had plans to integrate a few features in this vein so this is on the pile to consider. 
   2. The example data exemplifies an issue, however. If an event with a greater id completely encompasses an event with a smaller id, you will not see the eclipsed event unless the top event is moved. 
   3. Adding this feature will require also adding overlap detection and visualization. 
2. groupById
   1. This one I'm less keen on, as it requires giving the same identifier to multiple records. 
   2. Rather than co-opt ID, I'd look to adding instructions for a user adding custom properties and grouping by those. 
3. groupByIdDrawAllTitles
   1. This would go in the same category as groupById. 
   2. However, could see adding it commented out as an option should the user opt to add a custom grouping field. 

I'd be inclined at this time to pull the groupBySeries and hold on the other 2, potentially adding them as commented examples.

File: jquery.ganttView.css [hunk 1]
```diff
@@ -91,7 +91,7 @@ div.ganttview-slide-container {
 
 div.ganttview-grid-row-cell {
 	width: 20px;
-	height: 31px;
+	height: 100%;
```
Comment (lutze) [inline jquery.ganttView.css:94]: On my test implementation it looks like this change breaks the display.

**Comment by lutze** [comment]
I'll actually look at this for 0.9, please hold off on the rebase until then.

**Comment by GuillaumeSmaha** [comment]
Ok I will update it in few days.

**Comment by lutze** [comment]
I pulled this on a separate fork to review and merge. The support for the various grouping options balloons the addHeader / Block / etc functions significantly. I think a successful implementation of grouping could be simpler... Your method definitely works, but I'm going to have a look and see if there isn't another way to implement 2nd level grouping that might be a little more easily maintainable.

**Comment by iarts** [comment]
hello @lutze 
any news about multi-item lines yet?

**Comment by amff1704** [comment]
Hi @GuillaumeSmaha 

This would be great applied on Kanboard plugin “gantt-chart”. Using this to order tasks and show them grouped by column. (Using task’s expected time one could check total future occupied time in each column). This would be a great assets management tool applicable to a bunch of industries. 

Do you think it can be done (with reasonable amount of effort)? 


[https://kanboard.net/plugin/gantt](url)

**Pull request closed without merging**